Docu update (#562)
smart-data-lake/smart-data-lake: docs/reference/executionModes.md

  attribute1 = ...
}
```

> If no explicit executionMode is specified, the Default Behavior is used (see chapter below).

Reviewer: ```suggestion
If no explicit executionMode is specified, the default behavior is used (see chapter below).
```

There are 2 major types of Execution Modes selecting the subset of data based on:

* partitions 
* updates (incremental)

With partitions, the user selects the targeted partition or selecting all different partitions. 

> With incremental the difference between input and output can be selected. 

Reviewer: ```suggestion
With incremental, the difference between input and output can be selected. 
```

<!--TODO find a good introduction to partitions-->

### Default Behavior
A filter based on partitions can be applied manually by specifying the command line parameter `--partition-values` or `--multi-partition-values`, see [Command Line](commandLine.md). The partition values specified are passed to **all** start-Actions of a DAG and filtered for every input DataObject by its defined partition columns.
On execution every Action takes the partition values of the input and filters them again for every output DataObject by its defined partition columns, which serve again as partition values for the input of the next Action.
This can be used without providing any explicit Execution Mode in the config of your Actions. It is the Default Behavior of all Actions. 
Note that during execution of the dag, no new partition values are added, they are only filtered. An exception is if you place a `PartitionDiffMode` in the middle of your pipeline, see section [PartitionDiffMode](#partitiondiffmode-dynamic-partition-values-filter) below.
If the parameters `--partition-values` or `--multi-partition-values` are not specified, SDLB will process all available data.

### FailIfNoPartitionValuesMode

> The *FailIfNoPartitionValuesMode* enforces to specify partition values. It simply check if partition values are present and fail otherwise.

Reviewer: ```suggestion
The *FailIfNoPartitionValuesMode* enforces to specified partition values. It simply checks if partition values are present and fails otherwise.
```

> This is useful to prevent potential reprocessing of whole table through wrong usage.

Reviewer: ```suggestion
This is useful to prevent potential reprocessing of whole tables due to wrong usage.
```

### PartitionDiffMode: Dynamic partition values filter

> In contrast to specifying the partitions manually, you can let SmartDataLakeBuilder find missing partitions and set partition values automatically by specifying execution mode *PartitionDiffMode*. This mode has a couple of action to fine tune:

Reviewer: properties instead of action ?

Author: what about "options"?

Author: Funfact: This sentence was copy and pasted from the src comments and thus the description in the schema viewer. We should also adapt their...

By defining **alternativeOutputId** attribute you can define another DataObject which will be used to check for already existing data.
This can be used to select data to process against a DataObject later in the pipeline.

<!--TODO describe also PartitionDiffMode options  partitionColNb -->
<!--TODO describe also PartitionDiffMode options  nbOfPartitionValuesPerRun -->
<!--TODO describe also PartitionDiffMode options  applyPartitionValuesTransform -->
<!--TODO describe also PartitionDiffMode options  selectAdditionalInputExpression -->

### CustomPartitionMode

> An execution mode to create custom partition execution mode logic in scala.

Reviewer: .. that's quite difficult to understand.

Author: indeed. Will reformulate. Thanks.

By defining **alternativeOutputId** attribute you can define another DataObject which will be used to check for already existing data.
This can be used to select data to process against a DataObject later in the pipeline.

<!-- TODO describe DataFrameIncrementalMode -->
<!-- TODO describe FileIncrementalMode -->

## ProcessAllMode
An execution mode which forces processing all data from it's inputs, removing partitionValues and filter conditions received from previous actions.

## DataObjectStateIncrementalMode

> An execution mode for incremental processing by remembering DataObjects state from last increment. The state is saved in a .json file for which the user needs to provide a path using the `--state-path` option. An application name must also be specified with the `-n` option. 

Reviewer: ```suggestion
An execution mode for incremental processing by remembering DataObjects state from last increment. The state is saved in a .json file for which the user needs to provide a path using the `--state-path` option. An application name must also be specified with the `-n` option. The state file will be named after the application name.
```
